lib/rouge/lexers/livescript.rb

```diff
@@ -0,0 +1,308 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Livescript < RegexLexer
+      tag 'livescript'
+      aliases 'ls'
+      filenames '*.ls'
+      mimetypes 'text/livescript'
+
+      title 'LiveScript'
+      desc 'LiveScript, a language which compiles to JavaScript'
+
+      def self.detect?(text)
+        return text.shebang? 'lsc'
+      end
+
+      def self.declarations
+        @declarations ||= Set.new %w(const let var function class extends implements)
+      end
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          loop until for in of while break return continue switch case
+          fallthrough default otherwise when then if unless else throw try
+          catch finally new delete typeof instanceof super by from to til
+          with require do debugger import export yield
+        )
+      end
+
+      def self.constants
+        @constants ||= Javascript.constants + %w(yes no on off void)
+      end
+
+      def self.builtins
+        @builtins ||= Javascript.builtins + %w(this it that arguments)
+      end
+
+      def self.loop_control_keywords
+        @loop_control_keywords ||= Set.new %w(break continue)
+      end
+
+      id = /[$a-z_]((-(?=[a-z]))?[a-z0-9_])*/i
+      int_number = /\d+(_\d+)*/
+      int = /#{int_number}(e[+-]?#{int_number})?([A-Za-z_][A-Za-z0-9_]*)?/
```

Comment:
[The literals section](http://livescript.net/#literals) of the documentation seems to imply that it's only trailing letters. Is that correct? Assuming it is, my preference generally is for simpler patterns. Could we change it to:

```suggestion
      int = /#{int_number}(e[+-]?#{int_number})?([a-z]+)?/i
```

Reply:
Sorry it took so long, I have been very busy recently.

The regex was closer to the livescript transpiler implementation. Of course a feature that is not documented may actually not be a feature so it would be safer to rely on the docs. On the other hand livescript docs are too vague and does not mention some things e.g.:
- they say you can use dashes in variable names like in Lisp (`foo-bar`) and such names are converted to camel case (`fooBar`) but doesn't say that you cannot use numbers (`foo-bar-1`) because it is converted to an expression (`fooBar - 1`). https://github.com/gkz/LiveScript/issues/960
- they say nothing about scientific notation and starting from the first version you cannot use capital E in scientific notation because it is treated as a unit - `10E-1` is transpiled to `10 - 1`
- they say you can use letters as units but from the first version they use the same regex, which in fact is a bit simpler then the one I wrote and also allows `$`.

I think that changing the things mentioned above would be considered breaking changes in the language so it should be safe to use the following, simpler regex here (based on the lexer in Livescript repo):
```ruby
int = /#{int_number}(e[+-]?#{int_number})?[$\w]*/
```
What do you think?

Reply:
I think the LiveScript syntax sounds like an absolute mess :P But, given that, your solution looks good to me!